### PR TITLE
Revise CONTRIBUTING

### PR DESCRIPTION
**Pull request by thejohnfreeman**
These are changes I think belong in CONTRIBUTING. Let's have a discussion. There are some policy changes in here (two reviewers for all PRs) and some long-practiced-but-newly-stated polices added. I appreciate critique from a technical writing perspective as well.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (MarkusTeufelberger) [reply CONTRIBUTING.md:24]: Might also be worth having this checked by a bot in addition to stating it here.

File: CONTRIBUTING.md
Comment (MarkusTeufelberger) [inline CONTRIBUTING.md:26]: This might be worth keeping in, it should be easier to do the right thing than documenting only how to ignore the formatter. ;-)

Comment (thejohnfreeman) [reply CONTRIBUTING.md:26]: I tjhink that documenting how to use clang-format is outside the scope of this document, but I will add this example back in.

File: CONTRIBUTING.md
Comment (thejohnfreeman) [reply CONTRIBUTING.md:83]: No, because we have reviewers who are not maintainers, e.g. myself.

File: CONTRIBUTING.md
Comment (thejohnfreeman) [reply CONTRIBUTING.md:24]: I don't think GitHub's built-in protections can enforce this, and I'm not aware of a third-party bot that does it, but it is somewhat enforced by the fact that PRs cannot be squashed and merged unless all commits are signed.

Comment (ximinez) [reply CONTRIBUTING.md:24]: GitHub's protections can enforce that the tip of any branch is signed. I don't think it's necessary that _all_ commits are signed as long as the tip is. My understanding is that signing a commit is implicitly verifying all the commits leading up to it (sort of how validating a ledger in rippled implicitly validates all the previous ledgers). It may be worth rephrasing this to something like "All of your commits should be signed. The tip of pushed branches must be signed."

Comment (thejohnfreeman) [reply CONTRIBUTING.md:24]: I'll add this explanation to the document.

File: CONTRIBUTING.md
Comment (intelliot) [inline CONTRIBUTING.md:162]: @nbougalis - confirm the way you prefer your name? (if any preference)

**Review by intelliot — APPROVED** [review]

File: CONTRIBUTING.md
Comment (intelliot) [inline CONTRIBUTING.md:42]:
```suggestion
When you submit a PR for this change, please link the corresponding XLS draft in the description. As XLS drafts are considered work in progress and open for discussion, PRs should not be submitted without due time for discussions, questions, suggestions and additions to the XLS draft. It is the responsibility of the XLS author to update the draft to match the final implementation when the PR is merged.
```

Comment (intelliot) [reply CONTRIBUTING.md:42]: @thejohnfreeman this is precisely @Silkjaer's suggestion, just in the form of a GitHub "Suggested change" in case that is easier to see/adopt.

File: CONTRIBUTING.md
Comment (thejohnfreeman) [reply CONTRIBUTING.md]: I added this after some editing.

**Comment by intelliot** [comment]
will merge this today; if anyone has any issues with it, please open a new issue or PR with your desired changes

**Pull request merged**
